src/stdlib_io.fypp

```diff
@@ -236,17 +240,27 @@ contains
       !! call savetxt("log.txt", data)
       !!```
       !!
-
+      
       integer :: s, i, ios
+      character(len=1) :: delimiter_
+      character(len=3) :: delim_str
+      character(len=:), allocatable :: fmt_
       character(len=1024) :: iomsg, msgout
+
+      delimiter_ = optval(delimiter, " ")
```

Comment:
To avoid any potential issues in the future, I suggest to introduce the parameter `delimiter_default`
```fortran
character(len=1), parameter :: delimiter_default = " "
```

```suggestion
      delimiter_ = optval(delimiter, delimiter_default)
```